xarray/core/dataarray.py

```diff
@@ -6206,6 +6207,11 @@ def curvefit(
             this will be automatically determined by arguments of `func`. `param_names`
             should be manually supplied when fitting a function that takes a variable
             number of parameters.
+        allow_failures: bool, default: False
+            If True and the underlying `scipy.optimize_curve_fit` optimization fails for
+            any of the fits, return NaN in coefficients and covariances for those
+            coordinates. Helpful when fitting multiple curves and some of the data just
```

Comment:
Remove the Helpful... Part. It just adds clutter for little benefit.

If you want to add such tips do it above the argument list.

Reply:
Done in d3decbbd. I was also wondering if the name could be something different, like `allow_errors` for example. Are there any other examples of similar args in xarray, so we could be consistent with those?

Reply:
Good point.

In other places we use `errors = "raise" | "ignore" | "warn"` (the warn part is optional)

Reply:
Okay, I'll change it to that.

Reply:
Fixed in efce0b32. I was considering adding the option to also warn, but looks like scipy has its own internal mechanism for producing warnings, see here: https://github.com/scipy/scipy/blob/v1.10.1/scipy/optimize/_minpack_py.py#L491. I'm not sure how to control it and didn't want to get into that now. Maybe a warning is now always issued? Although I didn't see any warnings in pytest on my setup at least.